services/galley/src/Galley/API/Util.hs

```diff
@@ -466,22 +455,28 @@ fromRegisterConversation (Qualified usr localDomain) MkRegisterConversation {..}
         cnvReceiptMode = rcReceiptMode
       }
   where
-    me :: Map Domain [Public.Member] -> Galley Public.Member
-    me m = case Map.lookup localDomain m >>= find ((usr ==) . Member.memId) of
-      Nothing -> throwM convMemberNotFound
-      Just v -> pure v
-    others :: Map Domain [Public.Member] -> [OtherMember]
-    others =
-      Map.foldlWithKey' (\acc d mems -> fmap (memToOther d) mems <> acc) []
-        -- make sure not to include 'usr' in the list of others
-        . Map.adjust (filter ((usr /=) . Public.memId)) localDomain
-    memToOther :: Domain -> Member.Member -> OtherMember
-    memToOther d mem =
-      OtherMember
-        { omQualifiedId = Qualified (Member.memId mem) d,
-          omService = Member.memService mem,
-          omConvRoleName = Member.memConvRoleName mem
-        }
+    -- Currently this function creates a Member with default conversation attributes
+    -- FUTUREWORK(federation): retrieve member's conversation attributes (muted, archived, etc) here once supported by the database schema.
+    me :: Set OtherMember -> Galley Public.Member
+    me s =
+      case find ((== qusr) . omQualifiedId) . Set.toList $ s of
```

Comment:
This is an O(n) lookup. It's better to do the lookup on the set `s`.

Reply:
True, it is an `O(n)` lookup. Note I need to search via a projection function, and not by equality on `OtherMember`. I couldn't find a function to achieve this goal such that it wouldn't require `O(n)`. If you have an idea how, please let me know!

Reply:
Ah, I see. I guess there is no way at this point. But it should be fine if you implement my suggestion above. Then the whole thing is `O(n log n)` instead of `O(n^2)`.